hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HiveAvroSerializer.java

```diff
@@ -298,14 +297,12 @@ private Object serializePrimitive(PrimitiveObjectInspector fieldOI, Object struc
         String string = (String)fieldOI.getPrimitiveJavaObject(structFieldData);
         return new Utf8(string);
       case DATE:
-        return DateWritable.dateToDays(((DateObjectInspector)fieldOI).getPrimitiveJavaObject(structFieldData));
+        return new DateWritable((DateWritable)structFieldData).getDays();
```

Comment:
Hive3 will return DateWritableV2 for date type,  we cannot convert DateWritableV2 to DateWritable direclty.

Reply:
ohh, I compile with hive 3.1.2 and TestHiveAvroSerializer, the test is successful. Let me do more tests.